docs/docs/testing-css-in-js.md

```diff
@@ -0,0 +1,138 @@
+
+*Snapshot serializers* like `jest-styled-components` or `jest-emotion` modify the standard output to a more meaningful and readable snapshot, e.g. by removing unnecessary information or displaying data in another format. Which ultimately leads to more compareable and effective snapshot tests.
+
+By default snapshots of your styled components show the generated class names (which you didn't set) and no styling information. When changing the styles you'll only see the diff of some cryptic class names (hashes). That's why you should use the above mentioned *snapshot serializers*. They remove the hashes and format the CSS in style elements.
+
+For this example we'll use emotion. The testing utilities of emotion and glamor are largely based on [jest-styled-components](https://github.com/styled-components/jest-styled-components) so they have a similar usage. Please have a look at the testing section of your library to follow along.
+
+## Installation
+
+```sh
+npm install --save-dev jest-emotion babel-plugin-emotion
+```
+
+As [Gatsby's emotion plugin](https://www.gatsbyjs.org/packages/gatsby-plugin-emotion/) is using `babel-plugin-emotion` under the hood you'll also need to install it so that Jest can use it.
+
+If you followed along with the [Unit testing guide](docs/unit-testing) you'll have the file `jest-preprocess.js` at the root of your project. Open that file and add the plugin:
+
+```diff
+const babelOptions = {
+  presets: ["@babel/react", "@babel/env"],
+  plugins: [
++    "emotion",
+    "@babel/plugin-proposal-optional-chaining",
+    "@babel/plugin-proposal-class-properties",
+  ],
+}
+
+module.exports = require("babel-jest").createTransformer(babelOptions)
+```
+
+In order to tell Jest to use the serializer you'll need to create the file `setup-test-env.js` which will be run automatically before every test. Create the file `setup-test-env.js` at the root of your project. Insert this code into it:
+
+```js
+import { createSerializer } from 'jest-emotion';
+import * as emotion from 'emotion';
+
+expect.addSnapshotSerializer(createSerializer(emotion));
+```
+
+Lastly you need to tell Jest where to find this file. Open your `package.json` and add this entry to your `"jest"` section:
+
+```json
+"jest": {
+  "setupTestFrameworkScriptFile": "<rootDir>/setup-test-env.js"
+}
+```
+
+## Usage
+
+In this example you'll use `react-test-renderer` but you can also use [react-testing-library](docs/testing-react-components) or any other appropriate library.
+
+```js
```

Comment:
Where would someone paste this kind of code? What kind of situation calls for it, and what is an example of the type of file you'd use it in?

Reply:
Good point